frontend/src/components/files/FilesTable.tsx

```diff
@@ -28,138 +29,86 @@ const iconStyle = {
 };
 
 export default function FilesTable(props: FilesTableProps) {
-	const {publicView} = props;
-	// mapStateToProps
-	const filesInDataset = useSelector((state: RootState) => state.dataset.files);
-	const publicFilesInDataset = useSelector((state: RootState) => state.publicDataset.public_files);
-	const foldersInDataset = useSelector(
-		(state: RootState) => state.folder.folders
-	);
-	const publicFoldersIndataset =  useSelector(
-		(state: RootState) => state.folder.publicFolders
-	);
+	const { folderId, datasetId, filesInDataset, foldersInDataset, publicView } =
+		props;
+
 	// use history hook to redirect/navigate between routes
 	const history = useNavigate();
 	// get existing folder
-	const parentFolderId = props.folderId;
 	const selectFile = (selectedFileId: string | undefined) => {
 		// Redirect to file route with file Id and dataset id and folderId
 		history(
-			`/files/${selectedFileId}?dataset=${props.datasetId}&folder=${parentFolderId}&verNum=${selectedFileId}`
+			`/files/${selectedFileId}?dataset=${datasetId}&folder=${folderId}&verNum=${selectedFileId}`
 		);
 	};
 	const selectPublicFile = (selectedFileId: string | undefined) => {
 		// Redirect to file route with file Id and dataset id and folderId
 		history(
-			`/public/files/${selectedFileId}?dataset=${props.datasetId}&folder=${parentFolderId}&verNum=${selectedFileId}`
+			`/public/files/${selectedFileId}?dataset=${props.datasetId}&folder=${folderId}&verNum=${selectedFileId}`
 		);
 	};
 	const selectFolder = (selectedFolderId: string | undefined) => {
 		// Redirect to file route with file Id and dataset id
-		history(`/datasets/${props.datasetId}?folder=${selectedFolderId}`);
+		history(`/datasets/${datasetId}?folder=${selectedFolderId}`);
 	};
 
 	const selectPublicFolder = (selectedFolderId: string | undefined) => {
 		// Redirect to file route with file Id and dataset id
-		history(`/public/datasets/${props.datasetId}?folder=${selectedFolderId}`);
+		history(`/public/datasets/${datasetId}?folder=${selectedFolderId}`);
 	};
```

Comment:
@tcnichol I cleaned up your "publicView" logic in this component; also fix the bug of selectPublicFolder. Please double check if you have a chance.

Reply:
Should I do that on this pull request, or is there already a separate issue? Or does an issue need to be created for this?

Reply:
ah i fixed it in this branch.